src/liballoc/boxed.rs

```diff
@@ -883,6 +883,11 @@ impl<I: Iterator + ?Sized> Iterator for Box<I> {
     fn nth(&mut self, n: usize) -> Option<I::Item> {
         (**self).nth(n)
     }
+    default fn last(self) -> Option<I::Item> {
```

Comment:
Our current standard library policy is to not expose specialization in public APIs via `default`. Rather, we use an internal trait to mediate the specialization. Can you switch to that?

Also, the implementation could copy https://doc.rust-lang.org/nightly/src/core/iter/traits/iterator.rs.html#269.